spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/GroupingBatchBuilder.scala

```diff
@@ -28,11 +28,14 @@ import scala.collection.Iterator
  * @param data                  data iterator
  * @tparam T                    data type
  */
-class GroupingBatchBuilder[T](batchStatementBuilder: BatchStatementBuilder[T],
-                              batchKeyGenerator: BoundStatement => Any,
-                              batchSize: BatchSize,
-                              maxBatches: Int,
-                              data: Iterator[T]) extends AbstractIterator[RichStatement] with Iterator[RichStatement] {
+private[connector] class GroupingBatchBuilder[T](
+    boundStatementBuilder: BoundStatementBuilder[T],
+    batchStatementBuilder: BatchStatementBuilder,
+    batchKeyGenerator: BoundStatement => Any,
+    batchSize: BatchSize,
+    maxBatches: Int,
+    data: Iterator[T]) extends AbstractIterator[RichStatement] with Iterator[RichStatement] {
+
   require(maxBatches > 0)
```

Comment:
I know this isn't part of your change, but using require() without some context if it fails makes users sad